src/languages/ae/parser.mly

```diff
@@ -546,13 +546,13 @@ decl:
   | PRED p=raw_named_ident EQUAL body=lexpr
     { let loc = L.mk_pos $startpos $endpos in
       let ret_ty = T.prop ~loc () in
-      S.fun_def ~loc p [] [] ret_ty body }
+      S.pred_def ~loc p [] [] ret_ty body }
 
   | PRED p=raw_named_ident
     LEFTPAR args=separated_list(COMMA, logic_binder) RIGHTPAR EQUAL body=lexpr
     { let loc = L.mk_pos $startpos $endpos in
       let ret_ty = T.prop ~loc () in
-      S.fun_def ~loc p [] args ret_ty body }
+      S.pred_def ~loc p [] args ret_ty body }
```

Comment:
with the change to a dedicated function, you could remove the `ret_ty` from here and instead generate it in the body of the `pred_def` function.

Reply:
Done